### PR TITLE
#2 - Integration of CI build on AppVeyor.

### PR DESCRIPTION
**Pull request by maraf**
Fix project URL and tags in nuget package.
Publish GE.PM.nuget as AppVeyor artifact.
Publish GE.PM.zip as AppVeyor artifact.
Integrates SignPath hook on AppVeyor deploy.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- ps: .\tools\Prepare-Release.ps1
+
+test_script:
+- ps: .\tools\Run-Tests.ps1
+
+artifacts:
+- path: .\*.zip
+- path: .\*.nupkg
+
+deploy:
```
Comment (maraf) [inline appveyor.yml]: I'm not sure about SignPath, used whats in the main repository in release/3.1 branch.

File: src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj [hunk 2]
```diff
@@ -3,8 +3,8 @@
   <PropertyGroup>
     <TargetFramework>net461</TargetFramework>
     <Description>NuGet-based Plugin Manager for GitExtensions</Description>
-    <PackageTags>Neptuo GitExtensions</PackageTags>
-    <PackageProjectUrl>http://www.neptuo.com/project/dev/gitextensions.pluginmanager</PackageProjectUrl>
+    <PackageTags>GitExtensions</PackageTags>
+    <PackageProjectUrl>https://github.com/gitextensions/gitextensions.pluginmanager</PackageProjectUrl>
```
Comment (maraf) [inline src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj]: Will we be using github repository page or https://gitextensions.github.io?

File: src/PackageManager.UI/PackageManager.UI.csproj [hunk 3]
```diff
@@ -50,8 +50,4 @@
   </ItemGroup>
 
   <Import Project="$(MSBuildProjectDirectory)\..\..\build\Wpf.Build.targets" />
-
```
Comment (maraf) [inline src/PackageManager.UI/PackageManager.UI.csproj]: I don't see any value in publishing PM.exe as separate artifact here.

Comment (mast-eu) [reply src/PackageManager.UI/PackageManager.UI.csproj]: I agree

File: src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj (hunk 2 above)
Comment (mast-eu) [reply src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj]: I'd say the repo page. Doesn't match much sense to point from plugins to https://gitextensions.github.io/, which is only about the main app.

**Comment by RussKie** [comment]
We also have the same problem throughout our plugins codebase in the main
repo.
Something we need to get fixed too.

**Comment by maraf** [comment]
I have used totalcmd and searching through files and hopefully found all places where "GitExtensions" was not a name of anything in the code.

File: appveyor.yml (hunk 1 above)
Comment (RussKie) [reply appveyor.yml]: We will need to create a new signing policy, let's comment it out for now

File: README.md [hunk 4]
```diff
@@ -11,20 +11,20 @@ Nightly builds of PluginManager available at Appveyor - https://ci.appveyor.com/
 [![Build status](https://ci.appveyor.com/api/projects/status/k3y0frp1pgsyepwh?svg=true)](https://ci.appveyor.com/project/neptuo/gitextensions-pluginmanager)
 
 ### Browsing for packages
-PluginManager integrated into GitExtensions filters packages with dependency on `GitExtensions.Plugins`. This package is right now a kind of meta package and it is used to mark nuget packages intended as GitExtensions plugins.
+PluginManager integrated into Git Extensions filters packages with dependency on `GitExtensions.Plugins`. This package is right now a kind of meta package and it is used to mark nuget packages intended as Git Extensions plugins.
```
Comment (RussKie) [inline README.md]: Should we be changing  `GitExtensions.Plugins` to  `GitExtensions.Extensibility`?

Comment (maraf) [reply README.md]: Is the name final? I also mentioned `GitExtensions.Sdk` (like vsix dependency package) in discussion. For me personally all these names weight the same and it's only the decision.

Anyway, we also need to change the nuspec here and in the PM repo.

Comment (mast-eu) [reply README.md]: Personally, I like it simple: repo name = VS solution name = package name.
It requires less mental mapping.

Comment (maraf) [reply README.md]: > Personally, I like it simple: repo name = VS solution name = package name.
> It requires less mental mapping.

Definitely. I'm going to change the dependency..

**Comment by maraf** [comment]
If you are gentlemen ok with changes, I'm going to squash commits and merge.

@RussKie will you take care of AppVeyor project creation?

**Comment by mast-eu** [comment]
For me, this PR can be merged as soon as the AppVeyor project is set up.

**Comment by RussKie** [comment]
I have created two AppVeyor projects and sent you both invitations:
* https://ci.appveyor.com/project/gitextensions/gitextensions-extensibility
* https://ci.appveyor.com/project/gitextensions/gitextensions-pluginmanager

You should have sufficient permissions to control them. Please ping me directly if something needs tweaking.

**Comment by maraf** [comment]
@RussKie I crewed it up, I clicked on the link in the browser where I'm signed in with our organization account. Could you please kick the user I added and resend me another invitation?

**Comment by RussKie** [comment]
Done

**Comment by maraf** [comment]
We have a first successful CI build.

I had to add test packages as some tests use them. When I was migrating the original code I tough they were only used for manual testing, but failing tests reminded me that they had two purposes.

**Pull request merged**